### PR TITLE
Custom strategy for Tree

### PR DESCRIPTION
**Pull request by Smeagolworms4**
Add method for add custom strategy on TreeListener.

### CODE REVIEW

File: lib/Gedmo/Tree/TreeListener.php [hunk 1]
```diff
@@ -56,6 +56,18 @@ public function getSubscribedEvents()
         );
     }
 
+    /**
+     * @param straing  $name
```
Comment (NicoBoos) [inline lib/Gedmo/Tree/TreeListener.php]: You misspelled "string"

File: lib/Gedmo/Tree/TreeListener.php [hunk 2]
```diff
@@ -56,6 +56,18 @@ public function getSubscribedEvents()
         );
     }
 
+    /**
+     * @param string   $name
+     * @param Strategy $instance
+     * @return self
+     */
+    public function addStrategyInstance($name, $instance)
```
Comment (AkenRoberts) [inline lib/Gedmo/Tree/TreeListener.php]: What do you think about type-hinting `Gedmo\Tree\Strategy` here for `$instance`?

**Comment by Smeagolworms4** [comment]
You can merge ?

**Comment by franmomu** [comment]
can you please target `main` branch? parameters and return types should be added and also a test

**Pull request closed without merging**